examples/userapi/subdomains.ipynb

```diff
@@ -0,0 +1,954 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `initdamp` run in 0.04 s\n",
+      "Operator `padfunc` run in 0.01 s\n",
+      "Operator `padfunc` run in 0.01 s\n",
+      "Operator `padfunc` run in 0.01 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "so = 4 # FD space order (Note that the time order is by default 1).\n",
+    "\n",
+    "model = ModelElastic(space_order=so, vp=vp, vs=vs, rho=rho, origin=origin, shape=shape,\n",
+    "                     spacing=spacing, nbl=nbl, subdomains=(ur,lr))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now prepare model parameters along with the functions to be used in the PDE, and the source injection term:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s = model.grid.stepping_dim.spacing\n",
+    "\n",
+    "# Source freq. in MHz (note that the source is defined below):\n",
+    "f0 = 0.12\n",
+    "\n",
+    "# Thorbecke's parameter notation\n",
+    "l = model.lam\n",
+    "mu = model.mu\n",
+    "ro = model.irho\n",
+    "\n",
+    "t0, tn = 0., 30.\n",
+    "dt = 0.9*model.critical_dt\n",
+    "time_range = TimeAxis(start=t0, stop=tn, step=dt)\n",
+    "\n",
+    "# PDE functions:\n",
+    "v = VectorTimeFunction(name='v', grid=model.grid, space_order=so, time_order=1)\n",
+    "tau = TensorTimeFunction(name='t', grid=model.grid, space_order=so, time_order=1)\n",
+    "\n",
+    "# Source\n",
+    "src = RickerSource(name='src', grid=model.grid, f0=f0, time_range=time_range)\n",
+    "src.coordinates.data[:] = np.array([100., 50., 35.])\n",
+    "\n",
+    "# The source injection term\n",
+    "src_xx = src.inject(field=tau[0, 0].forward, expr=src*s)\n",
+    "src_yy = src.inject(field=tau[1, 1].forward, expr=src*s)\n",
+    "src_zz = src.inject(field=tau[2, 2].forward, expr=src*s)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now define the model equations. Sets of equations will be defined on each subdomain, 'upper' and 'lower', to which we pass the subdomain object when creating the equation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "u_v_u = Eq(v.forward, model.damp * (v + s*ro*div(tau)), subdomain = model.grid.subdomains['upper'])\n",
+    "u_t_u = Eq(tau.forward, model.damp * (tau + s * l * diag(div(v.forward))),\n",
+    "           subdomain = model.grid.subdomains['upper'])\n",
+    "\n",
+    "u_v_l = Eq(v.forward, model.damp * (v + s*ro*div(tau)), subdomain = model.grid.subdomains['lower'])\n",
+    "u_t_l = Eq(tau.forward, model.damp * (tau + s * l * diag(div(v.forward))\n",
+    "                                      + s * mu * (grad(v.forward) + grad(v.forward).T)),\n",
+    "           subdomain = model.grid.subdomains['lower'])"
```

Comment:
This isn't really acoustic water and elastic rock no? This does show nicely subdomain but it is still acoustic wave-equation with `mu=0` in the water so does the same as putting `0` in`mu.data`.

Also misleading that the intro talks about `05_staggered_acoustic.ipynb` which I thought would be the equation in the water

Reply:
Yes, I should clean this and the explanations up a bit (and remove mention of the staggered notebook).

As you mention, with the current setup the model being solved is the same as one not having subdomains since `mu` is zero in the water layer anyway. Originally I wanted to actually solve the acoustic equation as is done in the acoustic notebook and then form `tau` from `p` for the elastic eqn. But doing this 'nicely' wasn't really suitable for this tutorial (and should probably be in a different one?).

Reply:
Yes, it is clearer like that then adding interface conditions for the purpose of this tutorial, small text changes/cleanup nd should be gtg